### ISSUE TITLE
[A11Y FEATURE]: Allow user zoom

### ISSUE THREAD

**Issue by alvaromontoro** [jackdomleo7/Checka11y.css#154]
# A11y Feature Request

## Describe the new a11y feature

Some people need to zoom/increase the size of the page so they can read the text or see details in the pictures. Developers may disable that zoom functionality by using the "viewport" meta tag, and that can turn into an accessibility problem.

## Acceptance criteria

- [x] Show a warning if the viewport tag caps the zoom size (`maximum-scale=1`)
- [x] Show a warning if the viewport tag prevents users from zooming in/out (`user-scalable=no`)

### Possible solution

There is a way to do this related to the one suggested on #153. The idea would be to make the `<head>` tag visible (the elements inside are non-visible, but it could have some side effects), and display an error on the meta tag with the incorrect value. Something like this:

```css
/* this would be shared with #153 */
head {
  display: block;
}

head meta[name = "viewport"][content *= "maximum-scale=1" i],
head meta[name = "viewport"][content *= "user-scalable=no" i] {
  @include vectorMessage(error, "E0018", 458, "the display zoom has been", "disabled. Oh no!");
  display: block;
  border: 0;
}
```

## Why is this a11y feature important?

From [The A11Y Project](https://www.a11yproject.com/posts/never-use-maximum-scale/): "_Using the viewport meta tag the wrong way can prevent people with low vision concerns from accessing your website or web app_." In the article, the use of `maximum-scale` and `user-scalable=no` is discouraged. [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Viewport_meta_tag#viewport_basics) also includes a warning about the use of `user-scalable:no` and how it could have potential accessibility issues.

**Comment by jackdomleo7** [comment]
Thanks @alvaromontoro for this suggestion! I'd be happy to implement this, however I'm wondering if it should be a warning since there are some cases where web apps can disable zooming and with good reason. Unfortunately, though, can't think of any off the top of my head. Let me know what you think 😊

**Comment by alvaromontoro** [comment]
I think the maximum scale could be helpful when developing hybrid apps for mobile, to make it look more like a native app.

A warning should be fine for this.

**Comment by jackdomleo7** [comment]
Awesome, in which, I'm happy to go ahead with this feature!

**Comment by alvaromontoro** [comment]
Assign this ticket to me, please.
